Test

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+set -e
+
+trap 'cleanup; echo >&2 "FAILED!"' 0
+cleanup() { progress "Cleaning up"; kill $(jobs -p) >/dev/null ; }
+clean_exit() {
+    cleanup
+    trap '' 0
+    error_code=$(shift)
+    echo >&2 "$@"
+    exit $error_code
+}
+
+type pytest > /dev/null \
+    || clean_exit 1 -n \
+            'Please install pytest for python2 via pip.\n' \
+            'On ubuntu xenial run:: `pip2 install pip && pip2 install --user pytest`'
+
+progress() { echo ==== "$@" ; }
```

Comment:
line 5 uses progress. On my machine, running `./Test` throws `progress:Command not found`. Perhaps moving the definition above line 5 will fix this issue.